src/dioptra/restapi/v1/jobs/service.py

```diff
@@ -1055,3 +1204,68 @@ def create(
             )
 
         return {"mlflow_run_id": job.mlflow_run.mlflow_run_id}
+
+
+class ExperimentMetricsService(object):
+    """The service methods for retrieving metrics attached to jobs in the experiment."""
+
+    @inject
+    def __init__(
+        self,
+        experiment_jobs_service: ExperimentJobService,
+        job_id_metrics_service: JobIdMetricsService,
+    ) -> None:
+        """Initialize the experiment service.
+
+        All arguments are provided via dependency injection.
+        """
+        self._job_id_metrics_service = job_id_metrics_service
+        self._experiment_jobs_service = experiment_jobs_service
+
+    def get(
+        self,
+        experiment_id: int,
+        search_string: str,
+        page_index: int,
+        page_length: int,
+        sort_by_string: str,
+        descending: bool,
+        **kwargs,
+    ):
+        """Get a list of jobs and the latest metrics associated with each.
+
+        Args:
+            experiment_id: The unique id of the experiment.
+            error_if_not_found: If True, raise an error if the experiment is not found.
+                Defaults to False.
+
+        Returns:
+            The list of jobs and the metrics associated with them.
+
+        Raises:
+            EntityDoesNotExistError: If the experiment is not found and
+                `error_if_not_found` is True.
+        """
+        log: BoundLogger = kwargs.get("log", LOGGER.new())
+        log.debug(
+            "Get metrics for all jobs for an experiment by resource id",
+            resource_id=experiment_id,
+        )
+
+        jobs, num_jobs = self._experiment_jobs_service.get(
+            experiment_id,
+            search_string=search_string,
+            page_index=page_index,
+            page_length=page_length,
+            sort_by_string=sort_by_string,
+            descending=descending,
+            **kwargs,
+        )
+
+        job_ids = [job["job"].resource_id for job in jobs]
+
+        metrics_for_jobs = [
+            {"id": job_id, "metrics": self._job_id_metrics_service.get(job_id)}
```

Comment:
If even one of these is missing an mlflowRunId, an error is raised. This can happen when a job errors out before attaching an mlflowRunId.

Recommend adding a check for this and returning an empty list of metrics for that job.